diff -Nur libmcal.orig/Makefile.in libmcal/Makefile.in
--- libmcal.orig/Makefile.in	Tue Jan 28 18:45:21 2003
+++ libmcal/Makefile.in	Sat Apr 12 17:55:37 2003
@@ -7,23 +7,22 @@
 
 INCLUDE=@DRIVER_INCDIR@  ${OTHERINCDIR} -I.
 
-CFLAGS=-O0 -Wall -g ${INCLUDE} ${OTHER_CFLAGS}
+CFLAGS=$(OPTFLAGS) -Wall ${INCLUDE} ${OTHER_CFLAGS}
 
-LIBOBJS=mcal.o datetime.o lex.ical_yy.o icalroutines.o cal_misc.o $(DRIVER_LIBS)
+LIBOBJS=mcal.lo datetime.lo lex.ical_yy.lo icalroutines.lo cal_misc.lo $(DRIVER_LIBS)
 ALLOBJS=${LIBOBJS} tester.o
 OTHERLIBS=${OTHERLIBS}
 OTHERLIBDIR=${OTHERLIBDIR}
 
-LIBCAL=libmcal.a
-LIBCAL_SO=libmcal.so
+LIBCAL=libmcal.la
 
-all: libmcal.a libmcal.so
+all: libmcal.la
 
-install: libmcal.a libmcal.so
+install: libmcal.la
 	mkdir -p ${DESTDIR}@prefix@/lib/
 	mkdir -p ${DESTDIR}@prefix@/include/mcal/
-	install -m 755 -o 0 -g 0 libmcal.a libmcal.so ${DESTDIR}@prefix@/lib
-	install -m 644 -o 0 -g 0 \
+	libtool --mode=install install libmcal.la ${DESTDIR}@prefix@/lib
+	install -m 644 \
 	  mcal.h bool.h datetime.h drivers.h cal_misc.h icalroutines.h \
 	  ${DESTDIR}@prefix@/include/mcal/
 	touch installed
@@ -32,19 +31,16 @@
 	strip ${DESTDIR}@prefix@/lib/libmcal.so
 	strip ${DESTDIR}@prefix@/lib/libmcal.a
 
-libmcal.a: $(LIBOBJS)
+libmcal.la: $(LIBOBJS)
 	rm -f $(LIBCAL)
-	ar r $(LIBCAL) $(LIBOBJS)
+	libtool --mode=link $(CC) $(LDFLAGS) -o libmcal.la $(LIBOBJS) -rpath @prefix@/lib -lpam
 
-libmcal.so: $(LIBOBJS)
-	rm -f $(LIBCAL_SO)
-	gcc $(CFLAGS) -shared -o $(LIBCAL_SO) $(LIBOBJS)
 
 lex.ical_yy.c: icalscanner.lex
 	$(FLEX) $<
 
-tester: tester.o libmcal.a
-	gcc -Wall -g -o tester tester.o $(LIBCAL) -lcrypt
+tester: tester.o libmcal.la
+	libtool --mode=link $(CC) $(LDFLAGS) -Wall -o tester tester.o $(LIBCAL) -lcrypt
 
 clean:
 	rm -f *.o *~ core $(LIBCAL) $(LIBCAL_SO) tester installed 
@@ -56,5 +52,7 @@
 dep: depend
 
 depend:
-	makedepend $(INCLUDE) -- $(ALLOBJS:%.o=%.c) >& /dev/null
+	makedepend $(INCLUDE) -- $(ALLOBJS:%.lo=%.c) > /dev/null 2>&1
 
+%.lo: %.c
+	libtool --mode=compile $(CC) $(CFLAGS) -c $< -o $@
diff -Nur libmcal.orig/configure libmcal/configure
--- libmcal.orig/configure	Tue Jan 28 18:45:21 2003
+++ libmcal/configure	Sat Apr 12 17:58:01 2003
@@ -371,7 +371,7 @@
 	else
 	echo "checking for driver $ac_package... yes"
 	cat $ac_optarg/bootstrap.in >> bootstrap.in
-	echo "DRIVER_LIBS+=$ac_optarg/${ac_package}_driver.o" >> bootstrap.in
+	echo "DRIVER_LIBS+=$ac_optarg/lib${ac_package}_driver.la" >> bootstrap.in
 	echo  "#include \"$ac_optarg/${ac_package}.h\"" >> drivers.h
 	echo  "&${ac_package}_driver," >> drivers.c
 	DRIVER_INCDIR="-I$ac_optarg $DRIVER_INCDIR"
diff -Nur libmcal.orig/icap/Makefile libmcal/icap/Makefile
--- libmcal.orig/icap/Makefile	Tue Jan 28 18:31:33 2003
+++ libmcal/icap/Makefile	Sat Apr 12 17:55:53 2003
@@ -3,15 +3,15 @@
 CC=gcc
 FLEX=flex
 INCLUDE=-I.. -I../libmcal
-CFLAGS=-O0 -Wall -g $(INCLUDE)
-ALLOBJS=icap.o icaproutines.o lex.icap_yy.o
-TARGET=icap_driver.o
+CFLAGS=$(OPTFLAGS) -Wall $(INCLUDE)
+ALLOBJS=icap.lo icaproutines.lo lex.icap_yy.lo
+TARGET=libicap_driver.la
 
 all: $(TARGET)
 
 
 $(TARGET): $(ALLOBJS)
-	ld -r -o $(TARGET) $(ALLOBJS)
+	libtool --mode=link $(CC) -o $(TARGET) $(ALLOBJS)
 	touch bootstrap.in
 
 lex.icap_yy.c: icapscanner.lex
@@ -23,5 +23,7 @@
 dep: depend
 
 depend:
-	makedepend $(INCLUDE) -- $(ALLOBJS:%.o=%.c) >& /dev/null
+	makedepend $(INCLUDE) -- $(ALLOBJS:%.lo=%.c) > /dev/null 2>&1
 
+%.lo: %.c
+	libtool --mode=compile $(CC) $(CFLAGS) -c $< -o $@
diff -Nur libmcal.orig/mstore/Makefile libmcal/mstore/Makefile
--- libmcal.orig/mstore/Makefile	Tue Jan 28 18:31:33 2003
+++ libmcal/mstore/Makefile	Sat Apr 12 17:57:48 2003
@@ -4,15 +4,15 @@
 FLEX=flex
 INCLUDE=-I.. -I../libmcal
 # CFLAGS=-O0 -Wall -g $(INCLUDE) -DDEBUG -DUSE_PAM
-CFLAGS=-O0 -Wall -g $(INCLUDE) -DDEBUG
-ALLOBJS=mstore.o
-TARGET=mstore_driver.o
+CFLAGS=$(OPTFLAGS) -Wall $(INCLUDE) -DUSE_PAM
+ALLOBJS=mstore.lo
+TARGET=libmstore_driver.la
 
 all: $(TARGET)
 
 
 $(TARGET): $(ALLOBJS)
-	ld -r -o $(TARGET) $(ALLOBJS)
+	libtool --mode=link $(CC) -o $(TARGET) $(ALLOBJS)
 	touch bootstrap.in
 
 clean:
@@ -21,4 +21,7 @@
 dep: depend
 
 depend:
-	makedepend $(INCLUDE) -- $(ALLOBJS:%.o=%.c) >& /dev/null
+	makedepend $(INCLUDE) -- $(ALLOBJS:%.lo=%.c) > /dev/null 2>&1
+
+%.lo: %.c
+	libtool --mode=compile $(CC) $(CFLAGS) -c $< -o $@
